src/jdk.jfr/share/classes/jdk/jfr/consumer/RecordedStackTrace.java

```diff
@@ -51,7 +51,7 @@
     public List<RecordedFrame> getFrames() {
         Object[] array = getTyped("frames", Object[].class, null);
         if (array == null) {
-            return Collections.EMPTY_LIST;
+            return new ArrayList<>(0);
```

Comment:
Why ArrayList is used here and not List.of() as everywhere else?

Reply:
I wanted it to be consistent.

Now the list is always modifiable, regardless of how it is constructed. One could argue for always returning an  unmodifiable list, but it would require a spec change.